src/Extensions/README.md

```diff
@@ -0,0 +1,165 @@
+# Omex extension packages
+## Overview
+
+Packages created to provide Omex implementation of Extension interfaces that would be part of .NET 5 and currently distributed as nuget packges https://github.com/aspnet/Announcements/issues/411
+
+Packages use concept called Generic Host that is a set of interfaces for abstracting service run form infrastructure
+
+* `Microsoft.Omex.Extensions.Abstractions` - common primitives and utility classes, should be used for library creation
+* `Microsoft.Omex.Extensions.Logging` - implementation of `ILogger` interface that sends messages into our log aggregation system`
+* `Microsoft.Omex.Extensions.TimedScopes` - library for sending telemetry from `Activity` and `TimedScopes` to our log aggregation system
+* `Microsoft.Omex.Extensions.Compatibility` - provides access to some of the depricated static classes like `UlsLogger`, `Code.Validate`, `TimedScopes.Create`
+* `Microsoft.Omex.Extensions.Hosting` - package for cretion simple application (ex. Console app) based on Generic Host
+* `Microsoft.Omex.Extensions.Hosting.Services` - package for creation of Service Fabric services base on Generic Host
+* `Microsoft.Omex.Extensions.Hosting.Services.Web` - package for defining web listeners (based on Kestrel) in Service Fabric services
+* `Microsoft.Omex.Extensions.Hosting.Services.Remoting` - package for defining remoting listeners in Service Fabric services
+* `Microsoft.Omex.Extensions.Services.Remoting` - package for creating remoting listeners in Service Fabric
+
+## Goals
+* Isolation from Omex infrastructure
+* Isolation from Service Fabric
+* Using new Dependency Injection with resolution validation
+* Using null reference types
+* Replace `TimedScope` with `Activity`
+
+## Changed concepts
+### Logger
+Static `UlsLogger` replaced with `ILogger<TCategoryName>` that is taken from DI, genric type `TCategoryName` is usually the type where you are ingecting logger and name of the type would be used as logger category.
```

Comment:
>DI [](start = 77, length = 2)

please expand to Dependency Injection, alternatively please define it on first occurrence, e.g.

taken from Dependency Injection (DI)